src/java/main/org/apache/zookeeper/ClientCnxn.java

```diff
@@ -1231,14 +1231,14 @@ public void run() {
                             LOG.info(e.getMessage() + RETRY_CONN_MSG);
                         } else if (e instanceof RWServerFoundException) {
                             LOG.info(e.getMessage());
+                        } else if (e instanceof SocketException) {
+                            LOG.info("Socket error occurred: {}: {}", serverAddress, e.getMessage());
                         } else {
-                            LOG.warn(
-                                    "Session 0x"
-                                            + Long.toHexString(getSessionId())
-                                            + " for server "
-                                            + clientCnxnSocket.getRemoteSocketAddress()
-                                            + ", unexpected error"
-                                            + RETRY_CONN_MSG, e);
+                            LOG.warn("Session 0x{} for server {}, unexpected error{}",
```

Comment:
[non-blocking]

Although this isn't a regression, you probably want a ": " here (and a space); e.g.,

    "Session 0x{} for server {}, unexpected error: {}"

Reply:
No, because the RETRY_CONN_MSG should go right after it. (starts with ", ")